File: frontend/src/services/seat.service.ts
import { axiosInstance } from './api.client';
import type {
  Seat,
  SeatMapResponse,
  SeatReservationRequest,
  SeatReservationResponse,
} from '../types/seat.types';

export const seatService = {
  async getSeatMap(flightId: string): Promise<SeatMapResponse> {
    const response = await axiosInstance.get<SeatMapResponse>(`/seats/flight/${flightId}`);
    return response.data;
  },

  async getSeatById(seatId: number): Promise<Seat> {
    const response = await axiosInstance.get<Seat>(`/seats/${seatId}`);
    return response.data;
  },

  async reserveSeat(
    flightId: string,
    seatNumber: string,
    request: SeatReservationRequest
  ): Promise<SeatReservationResponse> {
    const response = await axiosInstance.post<SeatReservationResponse>(
      `/flights/${flightId}/seats/${seatNumber}/reserve`,
      request
    );
    return response.data;
  },

  async releaseSeat(flightId: string, seatNumber: string): Promise<void> {
    await axiosInstance.post(`/flights/${flightId}/seats/${seatNumber}/release`);
  },

  async confirmSeat(flightId: string, seatNumber: string): Promise<void> {
    await axiosInstance.post(`/flights/${flightId}/seats/${seatNumber}/confirm`);
  },
};

File: backend/src/main/java/com/skyhigh/service/SeatService.java
package com.skyhigh.service;

import com.skyhigh.dto.SeatMapResponseDTO;
import com.skyhigh.dto.SeatReservationResponseDTO;
import com.skyhigh.entity.Seat;

public interface SeatService {
    
    SeatMapResponseDTO getAvailableSeats(String flightId);
    
    Seat getSeatById(Long seatId);
    
    Seat getSeatByFlightAndNumber(String flightId, String seatNumber);
    
    SeatReservationResponseDTO reserveSeat(String flightId, String seatNumber, String passengerId);
    
    SeatReservationResponseDTO reserveSeatForCheckIn(String flightId, String seatNumber, String passengerId);
    
    Seat releaseSeat(Long seatId);
    
    Seat confirmSeat(Long seatId, String passengerId);
    
    Seat cancelSeat(Long seatId);
    
    int releaseExpiredSeats();
}
